grails-app/services/au/org/ala/bie/DownloadService.groovy

```diff
@@ -31,10 +30,11 @@ class DownloadService {
      */
     def download(params, OutputStream outputStream, Locale locale){
         def q = Encoder.escapeQuery(params.q ?: "*:*")
-        def fq = params.list('fq')
-        def fields = params.fields ?: grailsApplication.config.defaultDownloadFields ?: "guid,rank,scientificName,rk_genus,rk_family,rk_order,rk_class,rk_phylum,rk_kingdom,datasetName"
+        def fq = new ArrayList<>(params.list('fq')) // Make modifiable
+        def fields = params.fields ?: grailsApplication.config.defaultDownloadFields
```

Comment:
This would  be a good use for `grailsApplication.getRequiredProperty('defaultDownloadFields')` as it would throw a meaningful exception if the property was not set. 
FYI, when ever I change code that calls  `grailsApplication.foo.bar` I always use the utility ` `grailsApplication.getProperty('foo.isBar', Boolean, false)` style as it avoids a whole lot of runtime problems, such as auto-type coercion, NPE due to parent `foo` not being set, etc.

Reply:
I usually do that too but, in this case, since it's in application.yml it should never be blank. Changed it anyway for reasons of pureness and light.